src/assets/custom/js/websiteHeader.test.js

```diff
@@ -1,120 +1,139 @@
+      .spyOn(window, 'requestAnimationFrame')
       .mockImplementation(mockRequestAnimationFrame);
   });
 
   afterEach(() => {
     window.requestAnimationFrame.mockRestore();
   });
 
-  describe("When a page with a Website Header loads", () => {
+  describe('When a page with a Website Header loads', () => {
     beforeAll(() => {
       createMockHeader();
       simulatePageLoad();
-      header = window.document.querySelector(".website-header");
+      header = window.document.querySelector('.website-header');
     });
 
-    it("no special classes are applied", () => {
-      expect(header.classList).toContain("website-header");
-      expect(header.classList.length).toBe(1);
+    it("Should be displayed in it's revealed state", () => {
+      expect(header.classList).toContain('website-header');
+      expect(header.classList).toContain('website-header--revealed');
+      expect(header.classList.length).toBe(2);
     });
 
     describe("Given we have scrolled a long way down the page (and are a long way from the header's natural position)", () => {
       beforeEach(() => {
         simulateScrollingToY(1000);
       });
 
-      describe("When we then scroll up the page by just a couple of pixel", () => {
+      it('The header should be hidden', () => {
+        expect(header.classList).not.toContain('website-header--revealed');
+      });
+
+      describe('When we then scroll up the page by just a couple of pixel', () => {
         beforeEach(() => {
           simulateScrollingToY(1000 - 2);
         });
 
         it("we don't immediately reveal the header", () => {
-          expect(header.classList).toContain("website-header");
-          expect(header.classList).not.toContain("website-header--revealed");
-          expect(header.classList.length).toBe(1);
+          expect(header.classList).not.toContain('website-header--revealed');
         });
       });
 
-      describe("When we then scroll up the page by at least three pixels", () => {
+      describe('When we then scroll up the page by at least three pixels', () => {
         beforeEach(() => {
           simulateScrollingToY(1000 - 3);
         });
 
-        it("we reveal the header positioned at the top of the viewport", () => {
-          expect(header.classList).toContain("website-header");
-          expect(header.classList).toContain("website-header--revealed");
+        it('we reveal the header positioned at the top of the viewport', () => {
+          expect(header.classList).toContain('website-header');
+          expect(header.classList).toContain('website-header--revealed');
           expect(header.classList.length).toBe(2);
         });
 
-        describe("When we then scroll close to the top of the page", () => {
-          beforeEach(() => {
-            simulateScrollingToY(1);
+        describe('When we then open a submenu', () => {
+          beforeAll(() => {
+            header.classList.add(HEADER_HAS_OPEN_SUBMENU_CLASS);
```

Comment:
If you move the responsibility to know if a submenu is open out of this file, as per the previous comment/suggestion then you don't need to test both branches (submenu open + submenu not open). That would be simplier

Reply:
This has been addressed here: https://github.com/codurance/site/pull/1940/commits/d5ac023d4dc7a8f811b5b063a477250bf415ed33